docs/example_python.md

```diff
@@ -1,30 +1,194 @@
+local_resource(
+  'deploy',
+  'date +%s > start-time.txt'
+)
+…
+k8s_resource('example-python', port_forwards=5000, resource_deps=['deploy'])
+```
+
+The `local_resource()` call creates a local resource named `deploy`. The second
+argument is the command that it runs.
+
+See that button next to the `deploy` resource?
+![Button to trigger a deploy](assets/docimg/example-python-2-deploy-button.png)
+
+Let's click it and see what happens!
+
+[![Result of clicking the button  on the “deploy” resource](assets/docimg/example-python-3-naive-deploy.png)](https://cloud.tilt.dev/snapshot/AcyUjOYLaeKVxQ1XZjk=){:.is-image}
+*Clicking the button triggers the "deploy" local_resource, which in turn kicks off an update to the server. (Click the screenshot to see an interactive view.)*
+
+| Approach | Deploy Time (after initial)
+|---|---|
+| Naive | 10-11s |
+
+Can we do better?
+
+## Step 2: Let's Optimize It
+
+When we make a change to a file, we currently have to build an image, deploy new Kubernetes configs,
+and wait for Kubernetes to schedule the pod.
+
+With Tilt, we can skip all of these steps, and instead [live-update](https://docs.tilt.dev/live_update_tutorial.html) the pod in place.
+
+[Our new Tiltfile](https://github.com/windmilleng/tilt-example-html/blob/master/2-recommended/Tiltfile) contains the following new code:
+
+```python
+# Add live_update rules to our docker_build
+congrats = "🎉 Congrats, you ran a live_update! 🎉"
+docker_build('example-python-image', '.', live_update=[
+    sync('.', '/app'),
+    run('cd /app && pip install -r requirements.txt', trigger='/app/requirements.txt'),
+
+    # if all that changed was start-time.txt, make sure the server
+    # reloads so that it will reflect the new startup time
+    run('touch /app/app.py', trigger='start-time.txt'),
+
+    # add a congrats message!
+    run('sed -i "s/Hello cats!/{}/g" /app/templates/index.html'.format(congrats)),
+])
+```
+
+We've added a new parameter to `docker_build()` with four `live_update` steps that are, in order:
+1. Sync the code from the current directory (`.`) into the container at directory `/app`.
+2. IF `requirements.txt` has change, run `pip install`
+3. Poke `app.py` if necessary to make sure that Flask reloads the server
+4. Congratulation you on finishing this guide!
+
+Let's see what this looks like:
+
+[![Tilt state after a live_update](assets/docimg/example-python-4-liveupdate.png)](https://cloud.tilt.dev/snapshot/AdSLjOYLYo5KREvMpd4=){:.is-image}
+*The result of a live_update. (Click the screenshot to see an interactive view.)*
+
+Tilt was able to update the container in less than two seconds! (And a chunk of that time was overhead from Flask, not from Tilt.)
+
+## Our Recommendation
+
+### Final Score
+
+| Approach | Deploy Time (after initial)
+|---|---|
+| Naive | 10-11s |
+| With live_update | 1-2s |
+
+You can try the server here:
+
+[Recommended Structure](https://github.com/windmilleng/tilt-example-python/blob/master/2-live_update)
```

Comment:
I think it would be good to keep a convention of 'N-recommended' being the recommended workflow

Reply:
done